lib/exprengine.cpp

```diff
@@ -977,21 +977,41 @@ static bool isNonOverlapping(ExprEngine::ValuePtr v1, ExprEngine::ValuePtr v2)
 
 static bool conditionAlwaysFalse(ExprEngine::ValuePtr condValue, ExprEngine::DataBase *dataBase)
 {
-    if (auto v = std::dynamic_pointer_cast<ExprEngine::IntRange>(condValue))
-        return v->hasValue(0);
-    if (auto v = std::dynamic_pointer_cast<ExprEngine::FloatRange>(condValue))
-        return v->hasValue(0.0);
+    if (auto v = std::dynamic_pointer_cast<ExprEngine::IntRange>(condValue)) {
+        if (v->hasValue(0))
+            return true;
+        Data *data = dynamic_cast<Data *>(dataBase);
+        data->addConstraint(condValue, std::make_shared<ExprEngine::IntRange>("0", 0, 0), false);
+        return v->isEqual(dataBase, 0);
```

Comment:
Please do not modify data here.

I suggest that you check if the value can be != 0.. something like:

    return !ExprEngine::BinopValue("!=", v, zero).isTrue();